FileNotFoundError: [Errno 2] No such file or directory: 'h5diff'

### Please make sure these conditions are met

- [x] I have checked that this issue has not already been reported.
- [x] I have confirmed this bug exists on the latest version of anndata.
- [x] (optional) I have confirmed this bug exists on the main branch of anndata.

### Report

After cloning the repository

```
hatch test
```

(it takes 22 minutes)

and then I get an error:

```
FAILED tests/test_io_backwards_compat.py::test_no_diff[v0.11.4] -
FileNotFoundError: [Errno 2] No such file or directory: 'h5diff'
```


There were also a bunch of warnings. I copied a few as I am not sure there is a point copying all them here.


```
tests/test_concatenate.py::test_batch_key[var-xarray-concat-lazy]
  /home/gabor/work/anndata/tests/test_concatenate.py:1294: UserWarning: Variable names are not unique. To make them unique, call `.var_names_make_unique`.
    annot = get_annot(concat([lhs, rhs], axis=axis_name, force_lazy=force_lazy))
```

```
tests/test_concatenate.py::test_batch_key[var-xarray-concat-lazy]
  /home/gabor/work/anndata/tests/test_concatenate.py:1305: UserWarning: Variable names are not unique. To make them unique, call `.var_names_make_unique`.
    concat([lhs, rhs], axis=axis_name, label="batch", force_lazy=force_lazy)
```

```
tests/test_dask_view_mem.py::test_modify_view_mapping_component_memory[obsm-False]
  /home/gabor/work/anndata/tests/test_dask_view_mem.py:112: ImplicitModificationWarning: Trying to modify attribute `.obsm` of view, initializing view as actual.
    m[0, 0] = 100
```

```
tests/test_x.py: 1 warning
  /home/gabor/.local/share/hatch/env/virtual/anndata/88vkfPet/hatch-test.stable/lib/python3.13/site-packages/zarr/core/dtype/npy/string.py:248: UnstableSpecificationWarning: The data type (FixedLengthUTF32(length=4, endianness='little')) does not have a Zarr V3 specification. That means that the representation of arrays saved with this data type may change without warning in a future version of Zarr Python. Arrays stored with this data type may be unreadable by other Zarr libraries. Use this data type at your own risk! Check https://github.com/zarr-developers/zarr-extensions/tree/main/data-types for the status of data type specifications for Zarr V3.
    v3_unstable_dtype_warning(self)
```

```
docs/concatenation.rst::concatenation.rst
  <doctest concatenation.rst[45]>:1: UserWarning: Observation names are not unique. To make them unique, call `.obs_names_make_unique`.
```


### Versions

I don't think this is relevant


```
$ python3 -V
Python 3.13.7
```

08db5ccc43f6c21a0cda7bcbfcc51c4f58a3cef9

--- a/tests/test_io_backwards_compat.py
+++ b/tests/test_io_backwards_compat.py
@@ -1,5 +1,7 @@
 from __future__ import annotations
 
+import os
+import shutil
 from pathlib import Path
 from subprocess import run
 from typing import TYPE_CHECKING
@@ -49,6 +51,10 @@ def test_backwards_compat_files(archive_dir: Path) -> None:
     assert_equal(from_h5ad, from_zarr, exact=True)
 
 
+@pytest.mark.skipif(
+    not os.environ.get("CI") and shutil.which("h5diff") is None,
+    reason="not in CI and h5diff not installed",
+)
 def test_no_diff(tmp_path: Path, archive_dir: Path) -> None:
     if archive_dir.name in {"v0.7.8", "v0.7.0"}:
         pytest.skip("DataFrame encoding changed between 0.7 and now")

